lib/MusicBrainz/Server/Entity/Role/Age.pm

```diff
@@ -38,13 +38,16 @@ sub has_age
             )
         ) == -1;
 
-    # If there is no end date, then the end date is now() (so there is an age).
-    my @end_comp = $self->end_date->defined_run or return 1;
+    # If the entity is still active, the end date is now() (so there is an age).
+    return 1 if !$self->ended;
+
+    my @end_comp = $self->end_date->defined_run;
 
     # Shrink @begin_comp and @end_comp to the same size
-    my $shortest_run = min(scalar(@begin_comp) - 1, scalar(@end_comp) - 1);
-    @begin_comp = @begin_comp[0..$shortest_run];
-    @end_comp = @end_comp[0..$shortest_run];
+    my $shortest_run = min(scalar @begin_comp, scalar @end_comp);
+    return 0 if $shortest_run == 0; # no age if there is not even a year
+    splice @begin_comp, $shortest_run;
+    splice @end_comp, $shortest_run;
```

Comment:
The rest of the changes in this block is just for readability. I didn’t like the `-1`s, and the (0-based) index being called “shortest run”.

Arguably, `@begin_comp` and `@end_comp` still aren’t the best possible names, but …